src/app/api/v1/fxaRpEvents/route.ts

```diff
@@ -0,0 +1,212 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as jwt from 'jsonwebtoken'
+import jwkToPem from 'jwk-to-pem'
+import { NextRequest, NextResponse } from 'next/server'
+import { headers } from 'next/headers'
+import { captureException, captureMessage } from '@sentry/node'
+import { deleteSubscriber, getSubscriberByFxaUid, updateFxAProfileData, updatePrimaryEmail } from '../../../../db/tables/subscribers.js'
+import appConstants from "../../../../appConstants";
+
+const FXA_PROFILE_CHANGE_EVENT = 'https://schemas.accounts.firefox.com/event/profile-change'
+const FXA_PASSWORD_CHANGE_EVENT = 'https://schemas.accounts.firefox.com/event/password-change'
+const FXA_SUBSCRIPTION_CHANGE_EVENT = 'https://schemas.accounts.firefox.com/event/subscription-state-change'
+const FXA_DELETE_USER_EVENT = 'https://schemas.accounts.firefox.com/event/delete-user'
+
+/**
+ * Fetch FxA JWT Public for verification
+ *
+ * @returns {Promise<Array<jwt.JwtPayload> | undefined>} keys an array of FxA JWT keys
+ */
+const getJwtPubKey = async () => {
+  const jwtKeyUri = `${appConstants.OAUTH_ACCOUNT_URI}/jwks`
+  try {
+    const response = await fetch(jwtKeyUri, {
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+    const { keys } = await response.json()
+    console.info('getJwtPubKey', `fetched jwt public keys from: ${jwtKeyUri} - ${keys.length}`)
+    return keys
+  } catch (e) {
+    console.error('getJwtPubKey', `Could not get JWT public key: ${jwtKeyUri}`)
+    captureException(new Error(`Could not get JWT public key: ${jwtKeyUri} - ${e}`))
+  }
+}
+
+/**
+ * Authenticate FxA JWT for FxA relay event requests
+ *
+ * @param {NextRequest} req
+ * @returns {Promise<jwt.JwtPayload>} decoded JWT data, which should contain FxA events
+ */
+const authenticateFxaJWT = async (req: NextRequest) => {
+  // Assuming this is how you retrieve your auth header.
+  const requestHeaders = new Headers(req.headers)
+  requestHeaders.get('authorization')
+  const authHeader = requestHeaders.get('authorization')
+  
+  // Require an auth header
+  if (!authHeader) {
+    captureMessage(`No auth header found, ${requestHeaders.values()}`)
+    throw new Error('No auth header found');
+  }
+
+  // Extract the first portion which should be 'Bearer'
+  const headerType = authHeader.substring(0, authHeader.indexOf(' '))
+  if (headerType !== 'Bearer') {
+    captureMessage(`Invalid auth type, ${requestHeaders.values()}`)
+    throw new Error('Invalid auth type');
+  }
+
+  // The remaining portion, which should be the token
+  const headerToken = authHeader.substring(authHeader.indexOf(' ') + 1)
```

Comment:
Does `getToken()` from `"next-auth/jwt"` help here? I think it's supposed to parse the `Authorization` header and return the token if it is valid.

Reply:
If that doesn't help, it'd be nice to have this in middleware or a helper function.

Reply:
@rhelmer tried getToken, not seeing the token returned (getting a null). I think there might be some nextjs-specific code in there that stops the token from being parsed out.

Reply:
refactored